src/java.base/windows/classes/sun/nio/fs/WindowsFileSystemProvider.java

```diff
@@ -334,6 +334,18 @@ private void checkReadAccess(WindowsPath file) throws IOException {
                                 0L);
             fc.close();
         } catch (WindowsException exc) {
+            try {
+                if (exc.lastError() == ERROR_CANT_ACCESS_FILE) {
+                    FileChannel fc = WindowsChannelFactory
+                        .newFileChannel(file.getPathForWin32Calls(),
+                                file.getPathForPermissionCheck(),
+                                Set.of(WindowsChannelFactory.OPEN_REPARSE_POINT),
+                                0L);
+                    fc.close();
```

Comment:
checkAccess follows sym links and I wonder if there is any scenario where ERROR_CANT_ACCESS_FILE can be returned when respare point rather than the target cannot be accessed. This may be something the Microsoft folks can comment on.

Minor nit is the params to newFileChannel is are mis-aligned now.

Reply:
So, if the reparse point is a symbolic link, and if the target of the link does not exist, then  ERROR_FILE_NOT_FOUND is returned. I've tested and confirmed this. It's also what you would expect because you would expect usage of symbolic links to be transparent and the same error code to be returned as when traversing a path and some element of the path does not exist, or isn't accessible.

From what I can see ERROR_CANT_ACCESS_FILE can be returned for other reasons such as file corruption, which I think would be likely to return the same error again the second time.

I've also done some manual testing of other kinds of reparse point (junctions and mount points) and have tried various "target not existing" type of scenarios and ERROR_CANT_ACCESS_FILE is never returned.

Reply:
Though looking at that piece of code, I think its purpose would be clearer if it were put in a separate method with a name that shows were trying to open it as a socket.

Reply:
Moving it to a separate method would make it easier to maintain and discuss here but I would really prefer if it checked the reparse tag and re-throw the original exception if the tag is IO_REPARSE_TAG_SYMLINK. That way it wouldn't get tripped up by ERROR_CANT_ACCESS_FILE being thrown for other reasons, e.g. sym link -> sym link -> target where an intermediate sym link can't be accessed.

Reply:
Okay, I can add a check for WindowsFileAttributes.isUnixDomainSocket() in there when ERROR_CANT_ACCESS_FILE is returned.